frontend/catalyst/jax_tracer.py

```diff
@@ -932,3 +945,124 @@ def is_leaf(obj):
         # TODO: `check_jaxpr` complains about the `AbstractQreg` type. Consider fixing.
         # check_jaxpr(jaxpr)
     return closed_jaxpr, out_type, out_tree
+
+
+def null_postprocessing(results):
+    """A postprocessing function returned by a transform that only converts the batch of results
+    into a result for a single ``QuantumTape``.
+    """
+    return results[0]
+
+
+# pylint: disable=protected-access,no-member,not-callable
+def dynamic_one_shot(qnode):
+    """Transform a QNode to into several one-shot tapes to support dynamic circuit execution.
+
+    Args:
+        qnode (QNode): a quantum circuit which will run n-shot times
+
+    Returns:
+        qnode (QNode):
+
+        The transformed circuit to be run n-shot times such as to simulate dynamic execution.
+
+
+    **Example**
+
+    Consider the following circuit:
+
+    .. code-block:: python
+
+        dev = qml.device("lightning.qubit", shots=100)
+        params = np.pi / 4 * np.ones(2)
+
+        @qjit
+        @dynamic_one_shot
+        @qml.qnode(dev, diff_method=None)
+        def circuit(x, y):
+            qml.RX(x, wires=0)
+            m0 = measure(0, reset=reset, postselect=postselect)
+
+            @cond(m0 == 1)
+            def ansatz():
+                qml.RY(y, wires=1)
+
+            ansatz()
+            return measure_f(wires=[0, 1])
+
+    The ``dynamic_one_shot`` decorator prompts the QNode to perform a hundred one-shot
+    calculations, where in each calculation the ``measure`` operations dynamically
+    measures the 0-wire and collapse the state vector stochastically.
+    """
+
+    cpy_tape = None
+    aux_tapes = None
+
+    def transform_to_single_shot(qnode):
+        nonlocal cpy_tape
+        nonlocal aux_tapes
+        cpy_tape = None
+        aux_tapes = None
```

Comment:
Not too familiar with `nonlocal`, but if `transform_to_single_shot` is only called once (further down in this function), does that not make these overwrites redundant?

Reply:
Seems so, will change that.